### PR TITLE
[Perf]Remove H2D  for Qwen3.5 SpecV2

### PR DESCRIPTION
**Pull request by Chen-0210**
## Motivation

This PR improves Qwen3.5 specV2 performance by removing unnecessary H2D overhead in the `prepare_v2_verify` path.

## Modifications

1. Use `torch.stack(...).to(torch.int64)` to build `mamba_track_indices` in `eagle_info_v2.py`, avoiding the implicit scalar extraction path from a Python list of CUDA tensors.
2. Add a text-only fast path for SpecV2 mrope delta construction in `forward_batch_info.py`, so text-only verify batches can create the delta tensor directly on device.
## Accuracy Tests

Server
```
SGLANG_ENABLE_SPEC_V2=1 python -m sglang.launch_server \
  --model-path Qwen/Qwen3.5-397B-A17B/ \
  --port 30000 \
  --tp-size 8 \
  --speculative-algo NEXTN \
  --speculative-num-steps 3 \
  --speculative-eagle-topk 1 \
  --speculative-num-draft-tokens 4 \
  --mamba-scheduler-strategy extra_buffer
```
Benchmark
```
python3 benchmark/gsm8k/bench_sglang.py \
  --parallel 1319 \
  --num-questions 1319
```
Result
```
100%|██████████████████████████████████████████████████████████████████████████████████████████████| 1300/1300 [03:04<00:00, 7.05it/s]
Accuracy: 0.957
```

## Benchmarking and Profiling

Benchmark
`python3 -m sglang.bench_serving --backend sglang --dataset-name random --num-prompts 256 --random-input-len 1024 --random-output-len 1024 --random-range-ratio 1 --max-concurrency 4..64`
image

Profile
image

## Checklist

- [x] Format your code according to the [Format code with pre-commit](https://docs.sglang.io/developer_guide/contribution_guide.html#format-code-with-pre-commit).
- [x] Provide accuracy and speed benchmark results according to [Test the accuracy](https://docs.sglang.io/developer_guide/contribution_guide.html#test-the-accuracy) and [Benchmark the speed](https://docs.sglang.io/developer_guide/contribution_guide.html#benchmark-the-speed).
- [x] Follow the SGLang code style [guidance](https://docs.sglang.io/developer_guide/contribution_guide.html#code-style-guidance).

## Review Process

1. Ping Merge Oncalls to start the PR flow. See the [PR Merge Process](https://github.com/sgl-project/sglang/blob/main/.github/MAINTAINER.md#pull-request-merge-process).
2. Get approvals from [CODEOWNERS](https://github.com/sgl-project/sglang/blob/main/.github/CODEOWNERS) and other reviewers.
3. Trigger CI tests with [comments](https://docs.sglang.io/developer_guide/contribution_guide.html#how-to-trigger-ci-tests) or contact authorized users to do so.
   - `/tag-run-ci-label`, `/rerun-failed-ci`, `/tag-and-rerun-ci`
4. After green CI and required approvals, ask Merge Oncalls to merge.

### CODE REVIEW

**Review by Qiaolin-Yu — COMMENTED** [review]
Could you show the torch profiling result and benchmark result before and after the pr?

**Review by yizhang2077 — APPROVED** [review]
